File: factorial.c
/* Algorithm Technique: Recursion */
/* Finding factorial of a non negative integer */

#include<stdio.h>

unsigned long long factorial (int m);

int main(void)
{
    int a;

    printf("Enter number for which factorial is needed: ");
    scanf("%d", &a);

    printf("%d! is %llu\n", a, factorial(a));

    return 0;
}

unsigned long long factorial(int n)
{
    if (n == 0)
        return 1;
    else
        return n * factorial(n - 1);
}

File: Factorial.py
# Algorithmic technique: Recursion
# Finding a factorial of a natural number

def factorial(n):
    if n == 1:
        return 1
    else:
        n = n * factorial(n - 1)
    return n

x = input("Enter a natural number: ")
print("The factorial of the number "+ x + " is " + str(factorial(int(x))))
